ansible/configs/ansible-middleware-workshop/files/cloud_providers/ec2_cloud_template.j2

```diff
@@ -0,0 +1,375 @@
+#jinja2: lstrip_blocks: "True"
+---
+AWSTemplateFormatVersion: "2010-09-09"
+Mappings:
+  RegionMapping: {{ aws_ami_region_mapping | to_json }}
+
+Resources:
+  Vpc:
+    Type: "AWS::EC2::VPC"
+    Properties:
+      CidrBlock: "{{ aws_vpc_cidr }}"
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: "{{ aws_vpc_name }}"
+        - Key: Hostlication
+          Value:
+            Ref: "AWS::StackId"
+
+  VpcInternetGateway:
+    Type: "AWS::EC2::InternetGateway"
+
+  VpcRouteTable:
+    Type: "AWS::EC2::RouteTable"
+    Properties:
+      VpcId:
+        Ref: Vpc
+
+  VPCRouteInternetGateway:
+    DependsOn: VpcGA
+    Type: "AWS::EC2::Route"
+    Properties:
+      GatewayId:
+        Ref: VpcInternetGateway
+      DestinationCidrBlock: "0.0.0.0/0"
+      RouteTableId:
+        Ref: VpcRouteTable
+
+  VpcGA:
+    Type: "AWS::EC2::VPCGatewayAttachment"
+    Properties:
+      InternetGatewayId:
+        Ref: VpcInternetGateway
+      VpcId:
+        Ref: Vpc
+
+  PublicSubnet:
+    Type: "AWS::EC2::Subnet"
+    DependsOn:
+      - Vpc
+    Properties:
+    {% if aws_availability_zone is defined %}
+      AvailabilityZone: {{ aws_availability_zone }}
+    {% endif %}
+
+      CidrBlock: "{{ aws_public_subnet_cidr }}"
+      Tags:
+        - Key: Name
+          Value: "{{project_tag}}"
+        - Key: Hostlication
+          Value:
+            Ref: "AWS::StackId"
+      MapPublicIpOnLaunch: true
+      VpcId:
+        Ref: Vpc
+
+  PublicSubnetRTA:
+    Type: "AWS::EC2::SubnetRouteTableAssociation"
+    Properties:
+      RouteTableId:
+        Ref: VpcRouteTable
+      SubnetId:
+        Ref: PublicSubnet
+
+{% for security_group in security_groups|list + default_security_groups|list %}
+  {{security_group['name']}}:
+    Type: "AWS::EC2::SecurityGroup"
+    Properties:
+      GroupDescription: Host
+      VpcId:
+        Ref: Vpc
+      Tags:
+        - Key: Name
+          Value: "{{security_group['name']}}"
+{% endfor %}
+
+{% for security_group in default_security_groups|list + security_groups|list %}
+{% for rule in security_group.rules %}
+  {{security_group['name']}}{{rule['name']}}:
+    Type: "AWS::EC2::SecurityGroup{{rule['rule_type']}}"
+    Properties:
+     GroupId:
+       Fn::GetAtt:
+         - "{{security_group['name']}}"
+         - GroupId
+     IpProtocol: {{rule['protocol']}}
+     FromPort: {{rule['from_port']}}
+     ToPort: {{rule['to_port']}}
+  {% if rule['cidr'] is defined %}
+     CidrIp: "{{rule['cidr']}}"
+  {% endif  %}
```

Comment:
Watch out for the white space this will introduce.
```
{%   if rule['cidr'] is defined %}
     CidrIp: "{{rule['cidr']}}"
{%   endif  %}
```

Reply:
Thanks @jkupferer, this file was copied from the configs/three-tier-app config, would you like me to make these changes there also?